### PR TITLE
Convert to SDK exceptions for RuntimeBackend query executions

### LINKED ISSUE #726: Convert to SDK exceptions for `RuntimeBackend` query executions
do the same in https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/framework/crawlers.py#L152-L158 as we do in https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/mixins/sql.py#L105-L129

the most important exceptions are not found (schema, table), syntax errors, and PermissionDenied

### PR DESCRIPTION
**Pull request by pritishpai**
Convert the query execution exceptions to SDK exceptions like NotFound, PermissionDenied and SyntaxError (currently used BadRequest to map syntax errors).

Integration and unit tests added.

Fixes #726

### CODE REVIEW

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 1]
```diff
@@ -151,7 +157,18 @@ def __init__(self):
...
+            status = StatementStatus(state=StatementState.FAILED, error=ServiceError(message=str(e)))
+
+        if status is None:
+            status = StatementStatus(state=StatementState.FAILED)
+        if status.state == StatementState.SUCCEEDED:
+            return immediate_response
+
+        self._raise_if_needed(status)
 
     def fetch(self, sql) -> Iterator[Row]:
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:173]: We also need to wrap exceptions for fetch()

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 2]
```diff
@@ -151,7 +157,18 @@ def __init__(self):
 
     def execute(self, sql):
         logger.debug(f"[spark][execute] {sql}")
-        self._spark.sql(sql)
+        try:
+            immediate_response = self._spark.sql(sql)
+            status = immediate_response.status
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:162]: This is incorrect and too complicated. Don't involve statement status/etc. 

Keep it simple: Just use if ... in str(err): raise NotFound(...)

File: tests/integration/framework/test_crawlers.py [hunk 3]
```diff
@@ -9,3 +11,78 @@ def test_deploys_database(sql_backend, inventory_schema):
...
+    commands = CommandExecutor(ws.clusters, ws.command_execution, lambda: ws.config.cluster_id)
+
+    commands.install_notebook_library(f"/Workspace{wsfs_wheel}")
+    query_response_incorrect_schema = commands.run(
+        """
+        from databricks.labs.ucx.framework.crawlers import RuntimeBackend
+        from databricks.sdk.errors import NotFound
+        backend = RuntimeBackend()
+        try:
+            backend.execute("USE ABCDEF")
```
Comment (nfx) [inline tests/integration/framework/test_crawlers.py:26]: Use make_random() fixture to generate a name

**Review by nfx — CHANGES_REQUESTED** [review]
Implementation is incorrect and too complicated, simplify and make it also work with fetch() method

File: src/databricks/labs/ucx/framework/crawlers.py (hunk 1 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/framework/crawlers.py:174]: Done!

File: tests/integration/framework/test_crawlers.py (hunk 3 above)
Comment (pritishpai) [reply tests/integration/framework/test_crawlers.py]: Done! However could not do the same in unit tests as I got linting errors and make fmt removed the import statement.

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 4]
```diff
@@ -151,11 +157,28 @@ def __init__(self):
...
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:162]: There's no status in spark sql response, this is incorrect

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 5]
```diff
@@ -151,11 +157,28 @@ def __init__(self):
...
+        except Exception as e:
+            error_message = str(e)
+            self._raise_spark_sql_exceptions(error_message)
+
+        if status is None:
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:167]: Remove reliance on status from here altogether

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 6]
```diff
@@ -151,11 +151,28 @@ def __init__(self):
...
+        try:
+            immediate_response = self._spark.sql(sql)
+            # status = immediate_response.status
+        except Exception as e:
+            error_message = str(e)
+            self._raise_spark_sql_exceptions(error_message)
+
+        # if status is None:
+        #     status = StatementStatus(state=StatementState.FAILED)
+        # if status.state == StatementState.SUCCEEDED:
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:163]:
```suggestion
```
remove all commented come

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/framework/crawlers.py (hunk 6 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/framework/crawlers.py:163]: Yes. I just saw it. My bad!

Comment (pritishpai) [reply src/databricks/labs/ucx/framework/crawlers.py]: Done!

File: src/databricks/labs/ucx/framework/crawlers.py [hunk 7]
```diff
@@ -151,11 +151,22 @@ def __init__(self):
 
     def execute(self, sql):
         logger.debug(f"[spark][execute] {sql}")
-        self._spark.sql(sql)
+        try:
+            immediate_response = self._spark.sql(sql)
+            # status = immediate_response.status
```
Comment (nfx) [inline src/databricks/labs/ucx/framework/crawlers.py:156]:
```suggestion
```

**Review by nfx — APPROVED** [review]

**Pull request merged**